qiskit_experiments/calibration/fine_amplitude.py

```diff
@@ -0,0 +1,275 @@
+# that they have been altered from the originals.
+
+"""Fine amplitude calibration experiment."""
+
+from typing import List, Optional
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Gate
+from qiskit.qobj.utils import MeasLevel
+from qiskit.providers import Backend
+from qiskit.providers.options import Options
+from qiskit.pulse.schedule import ScheduleBlock
+
+from qiskit_experiments.base_experiment import BaseExperiment
+from qiskit_experiments.calibration.analysis.fine_amplitude_analysis import FineAmplitudeAnalysis
+from qiskit_experiments.exceptions import CalibrationError
+
+
+class FineAmplitude(BaseExperiment):
+    r"""Error amplifying fine amplitude calibration experiment.
+
+    The :class:`FineAmplitude` calibration experiment repeats N times a gate with a pulse
+    to amplify the under-/over-rotations in the gate to determine the optimal amplitude.
+    The circuits that are run have a custom gate with the pulse schedule attached to it
+    through the calibrations. The circuits are therefore of the form:
+
+    .. parsed-literal::
+
+                   ┌─────┐       ┌─────┐ ░ ┌─┐
+              q_0: ┤ Cal ├─ ... ─┤ Cal ├─░─┤M├
+                   └─────┘       └─────┘ ░ └╥┘
+        measure: 1/════════ ... ════════════╩═
+                                            0
+
+    Here, Cal is the name of the gate which will be taken from the name of the schedule.
+    The user can optionally add a square-root of X pulse before the Cal gates are repeated.
+    This square-root of X pulse allows the analysis to differentiate between over rotations
+    and under rotations in the case of pi-pulses. Importantly, the resulting data is analyzed
+    by a fit to a cosine function in which we try and determine the over/under rotation given
+    an intended rotation angle per gate which must also be specified by the user. The steps
+    to run a fine amplitude calibration experiment are therefore
+
+    .. code-block:: python
+
+        qubit = 3
+        amp_cal = FineAmplitude(qubit)
+        amp_cal.set_schedule(
+            schedule=x90p,
+            angle_per_gate=np.pi/2,
+            add_xp_circuit=False,
+            add_sx=False
+        )
+        amp_cal.run(backend)
+
+    Note that the schedule and angle_per_gate could have been set by independently calling
+    :meth:`set_experiment_options` for the schedule and :meth:`set_analysis_options` for
+    the angle_per_gate.
+
+    Error amplifying experiments are most sensitive to angle errors when we measure points along
+    the equator of the Block sphere. This is why users should insert a square-root of X pulse
+    before running calibrations for :math:`\pm\pi` rotations. Furthermore, when running
+    calibrations for :math:`\pm\pi/2` rotations users are advised to use an odd number of
+    repetitions, e.g. [1, 2, 3, 5, 7, ...] to ensure that the ideal points are on the equator

```

Comment:
Can we define something like `unit_sequence: int` and do (~L260)
```python
            for _ in range(repetition):
                for _ in range(unit_sequence):
                    circuit.append(gate, (0,))
```
? We can pass this number for fitter to divide estimated angle error per unit sequence. Actually, we can automatically infer this number by `pi/angle_per_gate` (by default). Then end users don't need to manage nrep.

Reply:
I'd like the user to be able to manage the repetitions as is currently implemented. I.e. change the number of repetitions as well as be able to change the increments. To make it easier to switch between X pulses and SX pulses I added two subclasses in 705e695 and amended the NB accordingly. This avoids having to mange the target angle and whether or not to add a SX gate before hand.